exporter/datadogexporter/internal/hostmetadata/internal/ec2/ec2.go

```diff
@@ -94,13 +109,13 @@ type Provider struct {
 }
 
 func NewProvider(logger *zap.Logger) (*Provider, error) {
-	sess, err := session.NewSession()
+	cfg, err := config.LoadDefaultConfig(context.Background())
 	if err != nil {
 		return nil, err
 	}
 	return &Provider{
 		logger:   logger,
-		detector: ec2provider.NewProvider(sess),
+		detector: ec2provider.NewProvider(cfg),
```

Comment:
The **datadogexporter** depends on the **NewProvider** function from **metadataproviders**, and the **resourcedetectionprocessor** also relies on the same function.